### PR TITLE
LPD-37883

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
Resend of [this](https://github.com/brianchandotcom/liferay-portal/pull/156142#issuecomment-2474703949)

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-api/src/main/java/com/liferay/commerce/product/constants/CPField.java [hunk 1]
```diff
@@ -46,6 +46,9 @@ public class CPField {
 	public static final String COMMERCE_OPTION_TYPE_KEY =
 		"commerceOptionTypeKey";
 
+	public static final String CP_CONFIGURATION_LIST_ID =
+		"cpConfigurationListId";
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-api/src/main/java/com/liferay/commerce/product/constants/CPField.java]: evaluate if having CP uppercase

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/query/contributor/CPConfigurationEntryKeywordQueryContributor.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+			keywordQueryContributorHelper.getSearchContext();
+
+		_queryHelper.addSearchTerm(
+			booleanQuery, searchContext, CPField.ASSET_CATEGORY_NAMES + ".text",
+			false);
+		_queryHelper.addSearchTerm(
+			booleanQuery, searchContext, CPField.EXTERNAL_REFERENCE_CODE,
+			false);
+		_queryHelper.addSearchTerm(
+			booleanQuery, searchContext, Field.ENTRY_CLASS_PK, false);
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/query/contributor/CPConfigurationEntryKeywordQueryContributor.java:44]: probably these should be sorted

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/result/contributor/CPConfigurationEntryModelSummaryContributor.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	public Summary getSummary(
+		Document document, Locale locale, String snippet) {
+
+		String prefix = Field.SNIPPET + StringPool.UNDERLINE;
+
+		String title = document.get(prefix + Field.NAME, Field.NAME);
+		String content = document.get(
+			prefix + Field.DESCRIPTION, Field.DESCRIPTION);
+
+		Summary summary = new Summary(title, content);
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/result/contributor/CPConfigurationEntryModelSummaryContributor.java:32]: declared as used..., good job! 😄

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CommerceCatalogLocalServiceImpl.java [hunk 4]
```diff
@@ -110,6 +116,29 @@ GroupConstants.DEFAULT_LIVE_GROUP_ID, getLocalizationMap(name),
...
+		if (calendar.get(Calendar.AM_PM) == Calendar.PM) {
+			displayDateHour += 12;
+		}
+
+		_cpConfigurationListLocalService.addCPConfigurationList(
+			null, commerceCatalog.getGroupId(), user.getUserId(), 0, true,
+			_language.format(
+				LocaleUtil.fromLanguageId(
+					commerceCatalog.getCatalogDefaultLanguageId()),
+				"master-configuration-x", commerceCatalog.getName(), false),
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CommerceCatalogLocalServiceImpl.java:136]: mmm... we are saving has name of the list `Master Configuration ${catalogName}`. Is there something that if the catalogName change also the name of the master configurations change?

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPConfigurationListLocalServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+		).given(
+			"There is a Commerce Catalog"
+		).when(
+			"A Configuration List is added"
+		).then(
+			"The Configuration List is created"
+		);
+
+		String externalReferenceCode = RandomTestUtil.randomString();
+		String listName = RandomTestUtil.randomString();
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPConfigurationListLocalServiceTest.java:91]: just name

File: modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPConfigurationListLocalServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+		).when(
+			"A child Configuration List is added"
+		).and(
+			"A different catalog is used to create it"
+		).then(
+			"An exception is thrown"
+		);
+
+		String externalReferenceCode1 = RandomTestUtil.randomString();
+		String listName1 = RandomTestUtil.randomString();
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPConfigurationListLocalServiceTest.java:151]: just name1

File: modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPConfigurationListLocalServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+
+		CPConfigurationList cpConfigurationList1 = _addCPConfigurationList(
+			externalReferenceCode1, _commerceCatalog.getGroupId(), 0, false,
+			listName1);
+
+		String externalReferenceCode2 = RandomTestUtil.randomString();
+		String listName2 = RandomTestUtil.randomString();
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPConfigurationListLocalServiceTest.java:158]: just name2

File: modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPConfigurationListLocalServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,213 @@
...
+
+		return _cpConfigurationListLocalService.addCPConfigurationList(
+			externalReferenceCode, groupId, _user.getUserId(),
+			parentCPConfigurationListId, masterCPConfigurationList, name, 0D,
+			calendar.get(Calendar.MONTH), calendar.get(Calendar.DAY_OF_MONTH),
+			calendar.get(Calendar.YEAR), displayDateHour,
+			calendar.get(Calendar.MINUTE), 0, 0, 0, 0, 0, true);
+	}
+
+	private static User _user;
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPConfigurationListLocalServiceTest.java:198]: why static?

File: modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPConfigurationEntryLocalServiceTest.java [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+
+		Assert.assertEquals(
+			cpConfigurationEntry1.getCPConfigurationEntryId(),
+			cpConfigurationEntry2.getCPConfigurationEntryId());
+	}
+
+	@Rule
+	public final FrutillaRule frutillaRule = new FrutillaRule();
+
+	private static User _user;
```
Comment (smottal) [inline modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPConfigurationEntryLocalServiceTest.java:130]: why static?

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-pricing-impl/src/main/java/com/liferay/headless/commerce/admin/pricing/internal/jaxrs/exception/mapper/RequiredBasePriceListExceptionMapper.java
Comment (smottal) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-pricing-impl/src/main/java/com/liferay/headless/commerce/admin/pricing/internal/jaxrs/exception/mapper/RequiredBasePriceListExceptionMapper.java:1]: this is the "old" way to do the Mappers... I know that you just renamed it and changed the exception but...

**Review by smottal — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-product-api/src/main/java/com/liferay/commerce/product/constants/CPField.java (hunk 1 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-api/src/main/java/com/liferay/commerce/product/constants/CPField.java]: change already

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/query/contributor/CPConfigurationEntryKeywordQueryContributor.java (hunk 2 above)
Comment (smottal) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/search/spi/model/query/contributor/CPConfigurationEntryKeywordQueryContributor.java:44]: No sorry, I saw now the CPField vs Field

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CommerceCatalogLocalServiceImpl.java (hunk 4 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CommerceCatalogLocalServiceImpl.java:136]: not sure to understand, I've just implemented the mockups

File: modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPConfigurationListLocalServiceTest.java (hunk 8 above)
Comment (andrea-ale-sbarra) [reply modules/apps/commerce/commerce-product-test/src/testIntegration/java/com/liferay/commerce/product/service/test/CPConfigurationListLocalServiceTest.java:198]: I think I've copied it somewhere

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Review by smottal — APPROVED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
ci:forward:force

**Pull request closed without merging**